tracer/test/Datadog.Trace.Security.Unit.Tests/IAST/OverheadControllerTests.cs

```diff
@@ -0,0 +1,118 @@
+        });
+        var instance = new OverheadController(new IastSettings(settings));
+        Assert.True(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        Assert.True(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+    }
+
+    [Fact]
+    public void GivenAnOverheadController_WhenSetSamplingTo100_AllOfRequestAreAcquired()
+    {
+        var settings = new CustomSettingsForTests(new Dictionary<string, object>()
+        {
+            { ConfigurationKeys.Iast.RequestSampling, 100 }
+        });
+        var instance = new OverheadController(new IastSettings(settings));
+        Assert.True(instance.AcquireRequest());
+        Assert.True(instance.AcquireRequest());
+        instance.ReleaseRequest();
+        Assert.True(instance.AcquireRequest());
+        instance.ReleaseRequest();
+        Assert.True(instance.AcquireRequest());
+    }
+
+    [Fact]
+    public void GivenAnOverheadController_WhenSetSamplingTo25_AQuarterOfRequestAreAcquired()
+    {
+        var settings = new CustomSettingsForTests(new Dictionary<string, object>()
+        {
+            { ConfigurationKeys.Iast.RequestSampling, 25 }
+        });
+        var instance = new OverheadController(new IastSettings(settings));
+        Assert.True(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+    }
+
+    [Fact]
+    public void GivenAnOverheadController_WhenSetMaxConcurrentRequestsTo1_Only1ConcurrentRequestsIsAllowed()
+    {
+        var settings = new CustomSettingsForTests(new Dictionary<string, object>()
+        {
+            { ConfigurationKeys.Iast.MaxConcurrentRequests, 1 },
+            { ConfigurationKeys.Iast.RequestSampling, 100 }
+        });
+        var instance = new OverheadController(new IastSettings(settings));
+        Assert.True(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        instance.ReleaseRequest();
+        Assert.True(instance.AcquireRequest());
+    }
+
+    [Fact]
+    public void GivenAnOverheadController_WhenSetMaxConcurrentRequestsTo2_Only2ConcurrentRequestsAreAllowed()
+    {
+        var settings = new CustomSettingsForTests(new Dictionary<string, object>()
+        {
+            { ConfigurationKeys.Iast.MaxConcurrentRequests, 2 },
+            { ConfigurationKeys.Iast.RequestSampling, 100 }
+        });
+        var instance = new OverheadController(new IastSettings(settings));
+        Assert.True(instance.AcquireRequest());
+        Assert.True(instance.AcquireRequest());
+        Assert.False(instance.AcquireRequest());
+        instance.ReleaseRequest();
+        Assert.True(instance.AcquireRequest());
+    }
+
+    [Fact]
+    public void GivenAnOverheadController_WhenWhenSetSamplingTo50AndSetMaxConcurrentRequestsTo2_ResultIsOk()
```

Comment:
Could you also add tests where the settings object has an invalid value (e.g. `0` or `101`) and the default value is used?

Reply:
I have added some IastSettingsTests with invalid values.